gpii/node_modules/preferencesServer/src/prefsSafesHandlers.js

```diff
@@ -0,0 +1,129 @@
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.defaults("gpii.preferencesServer.prefsSafeGet.handler", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            funcName: "gpii.preferencesServer.prefsSafeGet.handler.getPrefsSafe",
+            args: ["{gpii.preferencesServer.preferencesService}", "{preferencesServer}", "{that}", "{that}.req.params.prefsSafeId", "{that}.req.query.view"]
+        }
+    }
+});
+
+gpii.preferencesServer.prefsSafeGet.handler.getPrefsSafe = function (prefsService, preferencesServer, request, gpiiKey, view) {
+    var prefsPromise = prefsService.dataStore.findById(gpiiKey);
+    prefsPromise.then(request.events.onSuccess.fire, request.events.onError.fire);
+};
+
+fluid.defaults("gpii.preferencesServer.prefsSafeList.handler", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            funcName: "gpii.preferencesServer.prefsSafeList.handler.getPrefsSafeList",
+            args: ["{gpii.preferencesServer.preferencesService}", "{preferencesServer}", "{that}", "{that}.req.query.view"]
+        }
+    }
+});
+
+gpii.preferencesServer.prefsSafeList.handler.getPrefsSafeList = function (prefsService, preferencesServer, request, view) {
+    var finalDirectModel = fluid.extend(true, {}, prefsService.dataStore.listPrefsSafesDataSource.options.directModel, {});
+    var prefsPromise = prefsService.dataStore.listPrefsSafesDataSource.get(finalDirectModel);
+    prefsPromise.then(function (data) {
+        var togo = [];
+        // TODO this should probably be done in a couchview for efficiency and eventual paging?
+        fluid.each(data, function(item) {
+            item.value.prefsSafeId = item.key;
+            togo.push(item.value);
+        });
+        request.events.onSuccess.fire(togo);
+    }, request.events.onError.fire);
+};
+
+fluid.defaults("gpii.preferencesServer.listKeysForPrefsSafe.handler", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            funcName: "gpii.preferencesServer.listKeysForPrefsSafe.handler.getKeysForPrefsSafe",
+            args: ["{gpii.preferencesServer.preferencesService}", "{preferencesServer}", "{that}", "{that}.req.params.prefsSafeId", "{that}.req.query.view"]
+        }
+    }
+});
+
+gpii.preferencesServer.listKeysForPrefsSafe.handler.getKeysForPrefsSafe = function (prefsService, preferencesServer, request, prefsSafeId, view) {
+    var finalDirectModel = fluid.extend(true, {}, prefsService.dataStore.listKeysForPrefsSafeDataSource.options.directModel, {
```

Comment:
I don't see a value for `prefsService.dataStore.listKeysForPrefsSafeDataSource.options.directModel`? Would it be more straightforward to change this line to:

```
var directModel = {
    prefsSafeId: prefsSafeId
};
```